.github/workflows/ci.yml

```
@@ -0,0 +1,38 @@
+name: CI Pipeline
+
+on:
+    push:
+    pull_request:
+
+jobs:
+    ci-checks:
+        name: Run CI checks
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Repository
+              uses: actions/checkout@v4
+
+            - name: Install Rust
+              uses: dtolnay/rust-toolchain@stable
+              with:
+                components: clippy, rustfmt
+            
+            - name: Format Check
+              run: cargo fmt --check
+
+            - name: Build with Warnings Denied
+              run: RUSTFLAGS="-D warnings" cargo build --release
+
+            - name: Run Tests with Nextest
```

Don't specify the testing tool.
```suggestion
            - name: Cargo test
```